poor/util.py

```diff
@@ -273,6 +285,22 @@ def format_time(seconds):
         return _("{:d} min").format(minutes)
     return _("{:d} h {:d} min").format(hours, minutes)
 
+def round_distance(distance):
+    """Round distance taking into account used units and return it in meters"""
+    mile = 1609.34
+    yard = 0.9144
+    foot = yard / 3
+    if distance >= mile and (poor.conf.units == "american" or poor.conf.units == "british"):
+        return round(distance / mile, 1) * mile
+    elif poor.conf.units == "american":
+        return round(distance / foot / 100) * foot * 100
+    elif poor.conf.units == "british":
+        return round(distance / yard / 50) * yard * 50
+    # metric case left
+    elif distance > 1000:
+        return round(distance / 1e3, 1) * 1e3
+    return round(distance / 50) * 50
+
```

Comment:
Actually, I'm not sure if I understood this correctly or about the solution either. If it looks unclear or a bad idea, just skip it, and I'll look into it later.

Reply:
This function returns the distance that is simple to pronounce. Internally, all alert distances are in meters and, for each voice direction, we have a distance at which we say it (see where its called in narrative.py). So, depending on the unit, the distance will be slightly different to avoid saying 124 feet, for example.